satellite/satellitedb/certdb.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type certDB struct {
+	db *dbx.DB
+}
+
+func (certs *certDB) Set(ctx context.Context, nodeID storj.NodeID, pi *identity.PeerIdentity) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	tx, err := certs.db.Begin()
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+		} else {
+			err = errs.Combine(err, tx.Rollback())
+		}
+	}()
+
+	if pi == nil {
+		return Error.New("Peer Identity cannot be nil")
+	}
+	chain := identity.EncodePeerIdentity(pi)
+
+	var id []byte
+	query := `SELECT node_id FROM peer_identities WHERE serial_number = ?;`
+	err = tx.QueryRow(certs.db.Rebind(query), pi.Leaf.SerialNumber.Bytes()).Scan(&id)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			// create a new entry
+			_, err = tx.Exec(certs.db.Rebind(`INSERT INTO peer_identities ( serial_number, peer_identity, node_id, update_at ) VALUES ( ?, ?, ?, ? );`), pi.Leaf.SerialNumber.Bytes(), chain, nodeID.Bytes(), time.Now())
```

Comment:
If your objective is to insert only if the entry does not exist, you could simplify with
```
    INSERT INTO ... 
    ON CONFLICT (...)
    DO NOTHING
```

Reply:
@egonelbre  My understanding is, if Storagenode down the road changes its peer_identity certificate( and keep the same nodeID) then the new certification will be add as a chain to the previous ones and now the certificate serial number is different. If my understanding is, when it comes to make an entry in to the peer_identities table ( previously known as certDB) and assuming if previous entry exists, then the same row cannot be updated and intact a new entry is made and thus causing two entries for the same nodeID but different Certificate's serial number are created. If my understanding is correct, we should always maintain only one entry per node and just keep updating the nodeID's peer_id certificate if it changes, and changed peer id check is made using the certificate's serial number. Please advise. Currently implementation makes multiple entries.

Reply:
when storagenode can get a new leaf certificate, which should be store as a new entry. ie same
			storagenode can have multiple entries.But currently for a storagenode to get a new leaf identity, it is controlled in our alpha network. So the logic handles accordingly to make an entry if same serial-num exists.

Reply:
nit -- can we split this one really long line?

Reply:
Does "updated", meaning you split the line? Because it still looks like one line to me: https://github.com/storj/storj/pull/2764/files#diff-d2811cdba7bcdde3b1ebb1319da813a2R51